app/org/maproulette/framework/controller/TaskController.scala

```diff
@@ -48,4 +50,67 @@ class TaskController @Inject() (
       NoContent
     }
   }
+
+  /**
+    * Retrieve a task attachment
+    */
+  def attachment(id: Long, attachmentId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      this.sessionManager.userAwareRequest { implicit user =>
+        this.taskService.getTaskAttachment(id, attachmentId) match {
+          case Some(attachment) => Ok(attachment)
+          case None             => throw new NotFoundException(s"Attachment not found.")
+        }
+      }
+  }
+
+  /**
+    * Download the data from a task attachment as a file, decoded if necessary,
+    * and with the correct mime type
+    */
+  def attachmentData(id: Long, attachmentId: String, filename: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      this.sessionManager.userAwareRequest { implicit user =>
```

Comment:
Shouldn't this be an authenticated session, if you are downloading data to attach to task it probably should have a user associated with that action.

Reply:
The download request comes straight from JOSM, so it's not possible to send along the proper authentication data.

Reply:
Ah ok, that makes sense.